File: ClientApp/src/models/Homeowner.ts
import { Property } from './Property';
import { Payment } from './Payment';
import { Lien } from './Lien';
import { Fee } from './Fee';
import { Note } from './Note';

export interface Homeowner {
	id: number;
	propertyId: number;
	fullName: string;
	email: string;
	moveInDate: string;
	moveOutDate?: string;

	// virtual fields
	property: string;
}

export interface HomeownerDetails {
	// can't inherit bc type of property changes
	id: number;
	propertyId: number;
	fullName: string;
	email: string;
	moveInDate: string;
	moveOutDate?: string;

	property: Property;
	fees: Fee[];
	payments: Payment[];
	liens: Lien[];
	propertyNotes: Note[];
}

File: Models/Homeowner.cs
using Dapper.Contrib.Extensions;

namespace HoahServer.Models;

[Table("homeowners")]
public class Homeowner
{
    [Key] public int HomeownerId { get; set; }
    public int PropertyId { get; set; }
    public string FullName { get; set; } = "";
    public string? Email { get; set; }
    public string? Phone { get; set; }
    public string MoveInDate { get; set; } = "";
    public string? MoveOutDate { get; set; }

    [Write(false)] public string Property { get; set; } = "";
}

public class HomeownerDetails : Homeowner
{
    public new Property Property { get; set; } = new();
    public IEnumerable<Fee> Fees { get; set; } = new List<Fee>();
    public IEnumerable<Payment> Payments { get; set; } = new List<Payment>();
    public IEnumerable<Lien> Liens { get; set; } = new List<Lien>();
    public IEnumerable<Note> PropertyNotes { get; set; } = new List<Note>();
}